src/scenes/PosterPage/PosterPage.tsx

```diff
@@ -1,38 +1,55 @@
 import styled from 'styled-components';
 import { BaseM, BaseXL, TitleM } from 'components/core/Text/Text';
-import Button from 'components/core/Button/Button';
 import { contentSize, pageGutter } from 'components/core/breakpoints';
 import colors from 'components/core/colors';
 import PosterFigmaFrame from './PosterFigmaFrame';
 import ActionText from 'components/core/ActionText/ActionText';
 import StyledBackLink from 'components/NavbarBackLink/NavbarBackLink';
 import { useIsMobileWindowWidth } from 'hooks/useWindowSize';
-import { useToastActions } from 'contexts/toast/ToastContext';
 import useTimer from 'hooks/useTimer';
 import HorizontalBreak from 'components/core/HorizontalBreak/HorizontalBreak';
-import { MINT_DATE } from 'constants/poster';
-import InteractiveLink from 'components/core/InteractiveLink/InteractiveLink';
+import { MINT_DATE, NFT_TOKEN_ID } from 'constants/poster';
+import PosterMintButton from './PosterMintButton';
+import { GALLERY_MEMORABILIA_CONTRACT_ADDRESS } from 'hooks/useContract';
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers';
+import { useEffect, useState } from 'react';
+import { OPENSEA_TESTNET_BASEURL } from 'constants/opensea';
 
 export default function PosterPage() {
   const isMobile = useIsMobileWindowWidth();
-  const { pushToast } = useToastActions();
 
   const FIGMA_URL = 'https://www.figma.com/file/Opg7LD36QqoVb2JyOa4Kwi/Poster-Page?node-id=0%3A1';
-  const BRAND_POST_URL = 'https://gallery.mirror.xyz/1jgwdWHqYF1dUQ0YoYf-hEpd-OgJ79dZ5L00ArBQzac';
 
   const { timestamp, hasEnded } = useTimer(MINT_DATE);
+  const { account: rawAccount } = useWeb3React<Web3Provider>();
+  const account = rawAccount?.toLowerCase();
+  const [isMinted, setIsMinted] = useState(false);
 
   const handleBackClick = () => {
-    // TODO: Replace with hook
     window.history.back();
   };
 
-  const handleSignPoster = () => {
-    pushToast({
-      message: 'Thank you for participating in the (Object 006) 2022 Community Poster event.',
-      autoClose: true,
-    });
-  };
+  async function detectOwnedPosterNftFromOpensea(account: string) {
+    const response = await fetch(
+      `${OPENSEA_TESTNET_BASEURL}/api/v1/assets?owner=${account}&asset_contract_addresses=${GALLERY_MEMORABILIA_CONTRACT_ADDRESS}&token_ids=${NFT_TOKEN_ID}`,
```

Comment:
Might need to changed it when we go live

Reply:
good call👍👍